plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/repository/LibraryClasspathContainerResolverService.java

```diff
@@ -138,7 +138,33 @@ public IStatus resolveContainer(IJavaProject javaProject, IPath containerPath,
       return StatusUtil.error(this, Messages.getString("TaskResolveContainerError", containerPath), ex);
     }
   }
-  
+
+  public IStatus checkRuntimeAvailability(Runtime runtime, IProgressMonitor monitor) {
+    switch (runtime) {
+      case AppEngineStandard:
+      return checkAppEngineStandard(monitor);
+      default:
+        throw new IllegalArgumentException("Unhandled runtime: " + runtime);
+      }
+  }
+
+  private IStatus checkAppEngineStandard(IProgressMonitor monitor) {
```

Comment:
perhaps checkServletLibs or some such since this isn't really about App Engine per se.